ml-agents/mlagents/trainers/trainer/__init__.py

```diff
@@ -1 +1,2 @@
 from mlagents.trainers.trainer.trainer import Trainer  # noqa
+from mlagents.trainers.trainer.trainer_factory import TrainerFactory  # noqa
```

Comment:
I'd rather remove this and keep the import explicit:
```
from mlagents.trainers.trainer.trainer_factory import TrainerFactory
```

Reply:
Actually, why not just keep `trainer_factory.py` in the `trainers` directory?

Reply:
I am trying to compartmentalize. The TrainerFactory creates a Trainer, so I think it makes sense to have it in the trainers.trainer folder.

The only reason I put the import inside the init is because `mlagents.trainers.trainer.trainer_factory` is 3 times trainer :)